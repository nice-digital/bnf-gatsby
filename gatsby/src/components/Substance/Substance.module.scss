@use '@nice-digital/nds-core/scss/core';

.prepHeading {
  @include core.nds-h4;
  max-width: none;

  @include core.mq($from: md) {
    align-items: flex-start;
    display: flex;
    flex-direction: row-reverse;
    gap: core.rem(core.$nds-spacing-large);
  }
}

.headingIcons {
  align-items: center;
  display: flex;
  gap: core.rem(core.$nds-spacing-x-small);
  margin-bottom: core.rem(core.$nds-spacing-small);

  @include core.mq($from: md) {
    justify-content: flex-end;
    margin-bottom: 0;
    text-align: right;
  }
}

.headingText {
  display: block;
  margin-bottom: core.rem(core.$nds-spacing-small);

  @include core.mq($from: md) {
    flex-grow: 1;
  }
}

.label {
  @include core.nds-font-size(-2);
  background: core.$nds-colour-nice-dark-grey;
  border: 1px solid core.$nds-colour-border !important;
  color: core.$nds-colour-nice-white;
  display: inline-block;
  font-weight: 500;
  padding: core.rem(core.$nds-spacing-xx-small)
    core.rem(core.$nds-spacing-x-small);
  text-decoration: none !important;
  white-space: nowrap;
}

.alertContent {
  > :first-child {
    margin-top: 0;
  }

  > :last-child {
    margin-bottom: 0;
  }
}

.alert {
  display: block;

  @include core.mq($from: md) {
    flex-grow: 1;
  }
}
