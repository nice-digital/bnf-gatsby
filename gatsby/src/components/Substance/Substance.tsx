import { FC } from "react";

import { FeedBorderlineSubstance } from "@nice-digital/gatsby-source-bnf";
import { Alert } from "@nice-digital/nds-alert";

import { Accordion } from "@/components/Accordion/Accordion";
import { type QueryResult, type WithSlug } from "@/utils";

import Presentation from "./Presentation/Presentation";
import styles from "./Substance.module.scss";

export type SubstanceProps = {
	substance: WithSlug<QueryResult<FeedBorderlineSubstance>>;
	label?: string;
};

const Substance: FC<SubstanceProps> = ({
	substance: { title, slug, introductionNote, presentations },
	label,
}) => {
	return (
		<>
			<Accordion
				title={
					<>
						{introductionNote && (
							<Alert type="info">
								<div
									className={styles.alertContent}
									dangerouslySetInnerHTML={{ __html: introductionNote }}
								/>
							</Alert>
						)}
						<h2 id={slug} className={styles.prepHeading}>
							{label && (
								<span className={styles.headingIcons}>
									<span className={styles.label}>{label}</span>
								</span>
							)}
							<span
								className={styles.headingText}
								dangerouslySetInnerHTML={{ __html: title }}
							/>
						</h2>
					</>
				}
			>
				{presentations.map((presentation, i) => (
					<Presentation key={i} presentation={presentation}></Presentation>
				))}
			</Accordion>
		</>
	);
};

export default Substance;
