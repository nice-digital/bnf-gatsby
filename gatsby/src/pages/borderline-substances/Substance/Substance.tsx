import { FC } from "react";
import striptags from "striptags";

import { FeedBorderlineSubstance } from "@nice-digital/gatsby-source-bnf";
import { Alert } from "@nice-digital/nds-alert";

import { Accordion } from "@/components/Accordion/Accordion";
import { QueryResult } from "@/utils";

import styles from "./../{BnfBorderlineSubstancesTaxonomy.slug}.module.scss";
import Presentation from "./Presentation/Presentation";

export type SubstancesProps = {
	substance: QueryResult<FeedBorderlineSubstance>;
	label?: string;
};

const Substance: FC<SubstancesProps> = ({ substance, label }) => {
	return (
		<>
			<Accordion
				title={
					<>
						<h2
							key={substance.title}
							id={substance.id}
							className={styles.prepHeading}
						>
							<span className={styles.headingIcons}>
								{label ? <span className={styles.label}> {label} </span> : null}
							</span>
							<span className={styles.headingText}>{substance.title} </span>
						</h2>
						{substance && substance.introductionNote && (
							<Alert type="info">
								{striptags(substance?.introductionNote)}
							</Alert>
						)}
					</>
				}
			>
				{" "}
				{substance.presentations?.map((presentation, i) => (
					<Presentation key={i} presentation={presentation}></Presentation>
				))}
			</Accordion>
		</>
	);
};

export default Substance;
