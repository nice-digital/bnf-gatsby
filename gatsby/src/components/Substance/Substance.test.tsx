import { render, screen } from "@testing-library/react";

import Substance, { type SubstanceProps } from "./Substance";

const substanceProps: SubstanceProps = {
	substance: {
		title: "Fresubin® 1500 Complete",
		slug: "fresubin-1500-complete",
		presentations: [
			{
				acbs: [],
				carbohydrateGrams: "13.0",
				energyKCal: "100",
				energyKj: "420",
				fatGrams: null,
				fibreConstituents: [],
				fibreGrams: "1.5",
				formulation: "Liquid per 100 mL",
				presentationNote: null,
				proteinGrams: "3.8",
				fatConstituents: [],
				carbohydrateConstituents: [],
				proteinConstituents: [],
				borderlineSubstancePreps: [],
				rxAdvice: null,
				specialCharacteristics: [],
			},
		],
		introductionNote:
			"<p>Not suitable for use in child under 3 years; not recommended for child under 6 years</p>",
		id: "PHP103613",
	},
	label: "Some label",
};

describe("substance", () => {
	it("should match snapshot", () => {
		const { container } = render(<Substance {...substanceProps} />);

		expect(container).toMatchSnapshot();
	});

	it("should render a header for the substance", () => {
		render(<Substance {...substanceProps} />);
		const heading2 = screen.getByRole("heading", {
			level: 2,
		});

		// eslint-disable-next-line testing-library/no-node-access
		expect(heading2.parentElement).toHaveTextContent("Fresubin® 1500 Complete");
	});

	it("should render an alert containing the introduction note", () => {
		render(<Substance {...substanceProps} />);
		const alert = screen.getByText(
			"Not suitable for use in child under 3 years; not recommended for child under 6 years"
		);
		expect(alert).toBeTruthy;
	});

	it("should render a label", () => {
		render(<Substance {...substanceProps} />);
		const alert = screen.getByText("Some label");
		expect(alert).toBeTruthy;
	});

	it("should render each presentation", async () => {
		render(<Substance {...substanceProps} />);
		const presentation = screen.getAllByRole("definition");
		expect(presentation.length).toEqual(5);
		expect(presentation[0]).toHaveTextContent("Liquid per 100 mL");
	});
});
