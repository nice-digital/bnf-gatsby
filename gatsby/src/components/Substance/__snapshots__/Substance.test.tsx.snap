// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`substance should match snapshot 1`] = `
<div>
  <details
    class="details  "
  >
    <summary
      class="summary"
      data-tracking="Show"
    >
      <span
        class="toggleLabel label"
      >
        <svg
          aria-hidden="true"
          class="icon"
          fill="none"
          focusable="false"
          height="1em"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M478.152 174c0 10.091-3.525 18.661-10.576 25.712l-186 186c-7.243 7.243-15.909 10.864-26 10.864-10.283 0-18.853-3.621-25.712-10.864l-186-186C36.621 192.853 33 184.283 33 174c0-10.091 3.621-18.757 10.864-26L65 126.576C72.424 119.525 81.09 116 91 116c10.091 0 18.661 3.525 25.712 10.576L255.576 265.44 394.44 126.576c7.051-7.051 15.621-10.576 25.712-10.576 9.909 0 18.576 3.525 26 10.576L467.576 148c7.051 7.424 10.576 16.091 10.576 26z"
            fill="currentColor"
          />
        </svg>
        Show
      </span>
       
      <div
        class="alert alert--info"
      >
        <div
          class="alertContent"
        >
          <p>
            Not suitable for use in child under 3 years; not recommended for child under 6 years
          </p>
        </div>
      </div>
      <h2
        class="prepHeading"
        id="fresubin-1500-complete"
      >
        <span
          class="headingIcons"
        >
          <span
            class="label"
          >
            Some label
          </span>
        </span>
        <span
          class="headingText"
        >
          Fresubin® 1500 Complete
        </span>
      </h2>
    </summary>
    <dl>
      <div
        class="packDefinitionListItem"
      >
        <dt>
          Formulation
        </dt>
        <dd>
          Liquid per 100 mL
        </dd>
      </div>
      <div
        class="packDefinitionListItem"
      >
        <dt>
          Energy
        </dt>
        <dd>
          420
           kJ
           / 
          100
           kcal
        </dd>
      </div>
      <div
        class="packDefinitionListItem"
      >
        <dt>
          Protein
        </dt>
        <dd>
          3.8
           g
           
        </dd>
      </div>
      <div
        class="packDefinitionListItem"
      >
        <dt>
          Carbohydrate
        </dt>
        <dd>
          13.0
           g
           
        </dd>
      </div>
      <div
        class="packDefinitionListItem"
      >
        <dt>
          Fibre
        </dt>
        <dd>
          1.5
           g
           
        </dd>
      </div>
    </dl>
  </details>
</div>
`;
